### PR TITLE
Fix: Modify Tag path behavior in Dynamic Registries

### PR DESCRIPTION
**Pull request by rinOfTheStars**
Causes all modded Dynamic Registries to use `tags/NAMESPACE/path/to/file` for tag paths instead of `tags/path/to/file`, where `NAMESPACE` is the namespace of the Dynamic Registry. 

This prevents accidental namespace collision in tags, and could be considered a bugfix as it was intended behavior in the Dynamic Registry API to begin with.

### CODE REVIEW

File: library/data/tags/src/main/resources/quilt_tags.mixins.json
Comment (rinOfTheStars) [inline library/data/tags/src/main/resources/quilt_tags.mixins.json:1]: This is probably an artifact of an earlier implementation of the fix, can be 'removed' in a commit if desired to keep the number of changes lower, though it doesn't actually change any functionality.

Comment (Pyrofab) [reply library/data/tags/src/main/resources/quilt_tags.mixins.json:1]: Well I do think it's cleaner to have the mixins in order so no reason to revert it I'd say ?

**Comment by LambdAurora** [comment]
Could a tag be added for testing in the testmod?

**Comment by rinOfTheStars** [comment]
> Could a tag be added for testing in the testmod?

Done as of 3293df2

File: library/core/registry/src/main/java/org/quiltmc/qsl/registry/mixin/TagManagerLoaderMixin.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import net.minecraft.registry.tag.TagManagerLoader;
+import net.minecraft.util.Identifier;
+
+import org.quiltmc.qsl.registry.impl.dynamic.DynamicMetaRegistryImpl;
+
+@Mixin(TagManagerLoader.class)
+public class TagManagerLoaderMixin {
+	@Inject(method = "getRegistryDirectory", at = @At("HEAD"), cancellable = true)
+	private static void quilt$replaceModdedDynamicRegistryTagPath(RegistryKey<? extends Registry<?>> registry, CallbackInfoReturnable<String> cir) {
+		Identifier id = registry.getValue();
```
Comment (EnnuiL) [inline library/core/registry/src/main/java/org/quiltmc/qsl/registry/mixin/TagManagerLoaderMixin.java:35]: The review is pretty much going to be a tale of `var`s and newlines :p
```suggestion
		var id = registry.getValue();

```

File: library/core/registry/src/main/java/org/quiltmc/qsl/registry/mixin/TagManagerLoaderMixin.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package org.quiltmc.qsl.registry.mixin;
```
Comment (EnnuiL) [inline library/core/registry/src/main/java/org/quiltmc/qsl/registry/mixin/TagManagerLoaderMixin.java:17]: wait oh no, we have a missing license here;
Don't forget to run `./gradlew applyLicenses`! (dw, it's safe this time around)

File: library/core/registry/src/testmod/java/org/quiltmc/qsl/registry/test/dynamic/RegistryLibDynamicRegistryTest.java [hunk 3]
```diff
@@ -39,6 +45,7 @@ public class RegistryLibDynamicRegistryTest implements QuiltGameTest, ModInitial
 	private static final Greetings GREETING_A = new Greetings("Welcome to Quilt!", 5);
 	private static final Identifier GREETING_B_ID = new Identifier("quilt_registry_testmod", "greeting_b");
 	private static final Greetings GREETING_B = new Greetings("Howdy!", 2);
+	private static final Identifier GREETING_TEST_TAG_ID = new Identifier("quilt_registry_testmod", "test_tag");
```
Comment (EnnuiL) [inline library/core/registry/src/testmod/java/org/quiltmc/qsl/registry/test/dynamic/RegistryLibDynamicRegistryTest.java:48]: Oh, from what I saw in the code, you could directly create the TagKey here, having a `GREETING_TEST_TAG` object instead

File: library/core/registry/src/testmod/java/org/quiltmc/qsl/registry/test/dynamic/RegistryLibDynamicRegistryTest.java [hunk 4]
```diff
@@ -65,6 +72,23 @@ public void greetingsGetSynced(QuiltTestContext ctx) {
 		));
 	}
 
+	@GameTest(structureName = EMPTY_STRUCTURE)
+	public void greetingsTagGetLoaded(QuiltTestContext ctx) {
+		TagKey<Greetings> tagKey = TagKey.of(Greetings.REGISTRY_KEY, GREETING_TEST_TAG_ID);
+		Set<TagValues<Greetings>> tagValuesSet = TagRegistry.stream(Greetings.REGISTRY_KEY).collect(Collectors.toSet());
```
Comment (EnnuiL) [inline library/core/registry/src/testmod/java/org/quiltmc/qsl/registry/test/dynamic/RegistryLibDynamicRegistryTest.java:78]:
```suggestion
		var tagKey = TagKey.of(Greetings.REGISTRY_KEY, GREETING_TEST_TAG_ID);
		var tagValuesSet = TagRegistry.stream(Greetings.REGISTRY_KEY).collect(Collectors.toSet());

```

File: library/core/registry/src/testmod/java/org/quiltmc/qsl/registry/test/dynamic/RegistryLibDynamicRegistryTest.java [hunk 5]
```diff
@@ -65,6 +72,23 @@ public void greetingsGetSynced(QuiltTestContext ctx) {
...
+		ctx.failIf(() -> ctx.assertTrue(tagValuesSet.isEmpty(), "tagValuesSet should always be populated with at least 1 object"));
+
+		ctx.succeedIf(() -> ctx.assertTrue(tagValuesSet.stream().anyMatch(tagValues -> {
+			Registry<Greetings> greetingsRegistry = ctx.getWorld().getRegistryManager().get(Greetings.REGISTRY_KEY);
```
Comment (EnnuiL) [inline library/core/registry/src/testmod/java/org/quiltmc/qsl/registry/test/dynamic/RegistryLibDynamicRegistryTest.java:82]: Yes, the original code had one of those too, I would change the other instance if I was me :p
```suggestion
			var greetingsRegistry = ctx.getWorld().getRegistryManager().get(Greetings.REGISTRY_KEY);
```

File: library/core/registry/src/testmod/java/org/quiltmc/qsl/registry/test/dynamic/RegistryLibDynamicRegistryTest.java [hunk 6]
```diff
@@ -65,6 +72,23 @@ public void greetingsGetSynced(QuiltTestContext ctx) {
...
+			Greetings testerA = greetingsRegistry.getOrEmpty(GREETING_A_ID).orElse(null);
```
Comment (EnnuiL) [inline library/core/registry/src/testmod/java/org/quiltmc/qsl/registry/test/dynamic/RegistryLibDynamicRegistryTest.java:83]:
```suggestion
			var testerA = greetingsRegistry.getOrEmpty(GREETING_A_ID).orElse(null);
```
although hm, thinking about it, maybe this bit could be simplified a bit? i'll tinker with the code once it's merged

File: library/core/registry/src/testmod/java/org/quiltmc/qsl/registry/test/dynamic/RegistryLibDynamicRegistryTest.java [hunk 7]
```diff
@@ -65,6 +72,23 @@ public void greetingsGetSynced(QuiltTestContext ctx) {
...
+			if (Objects.isNull(testerA)) return false;
+
+			Set<Greetings> heldIds = new HashSet<>();
+			tagValues.values().iterator().forEachRemaining(greetingsHolder -> heldIds.add(greetingsHolder.value()));
+			return tagValues.key().equals(tagKey) && heldIds.contains(testerA);
```
Comment (EnnuiL) [inline library/core/registry/src/testmod/java/org/quiltmc/qsl/registry/test/dynamic/RegistryLibDynamicRegistryTest.java:88]:
```suggestion
			var heldIds = new HashSet<>();
			tagValues.values().iterator().forEachRemaining(greetingsHolder -> heldIds.add(greetingsHolder.value()));

			return tagValues.key().equals(tagKey) && heldIds.contains(testerA);
```

**Review by EnnuiL — CHANGES_REQUESTED** [review]
Other than style nitpicks (and uh, maybe a heckton of them :p), the only thing blocking this PR is really a missing license header; Otherwise, this PR can be merged and fast-tracked

**Review by EnnuiL — APPROVED** [review]
While I still have style nitpicks, I'll unblock this PR now that it compiles :p

**Review by LemmaEOF — APPROVED** [review]

**Review by EnnuiL — APPROVED** [review]
don't invalidate my reviews ;-;

**Pull request merged**